typing/rec_check.ml

```diff
@@ -169,7 +169,7 @@ let is_ref : Types.value_description -> bool = function
 let scrape env ty =
   (Ctype.repr (Ctype.expand_head_opt env (Ctype.correct_levels ty))).desc
```

Comment:
It's not part of your diff, but that call to `repr` is useless I believe. (Though it doesn't cost much and makes it easier to reason about the code, so perhaps it's fine to just leave it here).